components/mcp3008rst

```diff
@@ -0,0 +1,70 @@
+MCP3008 I/O Expander
+=====================
+
+.. seo::
+    :description: Instructions for setting up MCP3008 10 Bit Analog to Digital Converter in ESPHome.
+    :image: mcp3008.png
+
+The Microchip Technology Inc. MCP3008
+devices are successive approximation 10-bit Analogto-Digital (A/D) converters with on-board sample and
+hold circuitry
```

Comment:
You should say somewhere what the value read on the pins are? Which is the voltage reference for the sampling? Etc.